src/wheel/_wheelfile.py

```diff
@@ -17,17 +17,19 @@
 from os import PathLike
 from pathlib import Path, PurePath
 from types import TracebackType
-from typing import IO, TYPE_CHECKING, BinaryIO, NamedTuple, cast
+from typing import IO, BinaryIO, NamedTuple, Tuple, cast
 from zipfile import ZIP_DEFLATED, ZIP_STORED, ZipFile, ZipInfo
 
 from . import __version__ as wheel_version
-from .vendored.packaging.utils import InvalidWheelFilename, parse_wheel_filename
+from .vendored.packaging.tags import Tag
+from .vendored.packaging.utils import (
+    InvalidWheelFilename,
+    NormalizedName,
+    Version,
+    parse_wheel_filename,
+)
 
-if TYPE_CHECKING:
-    from packaging.tags import Tag
-    from packaging.utils import NormalizedName, Version
-
-    WheelContentElement = tuple[tuple[PurePath, str, str], BinaryIO]
+WheelContentElement = Tuple[Tuple[PurePath, bytes, int], BinaryIO]
```

Comment:
If this is going be part of the public API then it shouldn't be in `TYPE_CHECKING`.  It'd make it a little hard to work with the type outside of a static type checker.

I've also thought it could be NamedTuple.

Reply:
You are right about that. I also need to re-export it in `__init__.py`.